### PR TITLE
netlink: introduce sd_netlink_message_has_flag()

### PR DESCRIPTION
**Pull request by yuwata**
Split out of #18589.

### CODE REVIEW

File: src/libsystemd/sd-netlink/netlink-message.c [hunk 1]
```diff
@@ -686,7 +685,8 @@ static int netlink_message_read_internal(sd_netlink_message *m, unsigned short t
 
         rta = (struct rtattr*)((uint8_t *) m->hdr + attribute->offset);
 
-        *data = RTA_DATA(rta);
+        if (data)
+                *data = RTA_DATA(rta);
```
Comment (poettering) [inline src/libsystemd/sd-netlink/netlink-message.c]: maybe return the two params ret_data and ret_net_byteorder even

Comment (yuwata) [reply src/libsystemd/sd-netlink/netlink-message.c]: Hm? Sorry, what do you mean?

Comment (yuwata) [reply src/libsystemd/sd-netlink/netlink-message.c]: Ah, you mean that the variables should be renamed, don't you?

Comment (poettering) [reply src/libsystemd/sd-netlink/netlink-message.c]: yupp. 

meant to write "rename" when i wrote "return". sorry for the confusion

**Comment by poettering** [comment]
change lookgs good to me. but CI is a bit unhappier than usual

**Comment by yuwata** [comment]
I restarted the failed CIs and now they look fine.

**Pull request merged**